server/src/internalClusterTest/java/org/elasticsearch/indices/SystemIndexManagerIT.java

```diff
@@ -0,0 +1,255 @@
+import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertAcked;
+import static org.hamcrest.Matchers.equalTo;
+
+@ESIntegTestCase.ClusterScope(scope = ESIntegTestCase.Scope.TEST, numDataNodes = 0)
+public class SystemIndexManagerIT extends ESIntegTestCase {
+
+    private static final String INDEX_NAME = ".test-index";
+    private static final String PRIMARY_INDEX_NAME = INDEX_NAME + "-1";
+
+    @Before
+    public void beforeEach() {
+        TestSystemIndexDescriptor.useNewMappings.set(false);
+    }
+
+    @Override
+    protected Collection<Class<? extends Plugin>> nodePlugins() {
+        return CollectionUtils.appendToCopy(super.nodePlugins(), TestPlugin.class);
+    }
+
+    /**
+     * Check that if the the SystemIndexManager finds a managed index with out-of-date mappings, then
+     * the manager updates those mappings.
+     */
+    public void testSystemIndexManagerUpgradesMappings() throws Exception {
+        internalCluster().startNodes(1);
+
+        // Trigger the creation of the system index
+        assertAcked(prepareCreate(INDEX_NAME));
+        ensureGreen(INDEX_NAME);
+
+        assertMappings(TestSystemIndexDescriptor.getOldMappings());
+
+        // Poke the test descriptor so that the mappings are now "updated"
+        TestSystemIndexDescriptor.useNewMappings.set(true);
+
+        // Cause a cluster state update, so that the SystemIndexManager will update the mappings in our index
+        triggerClusterStateUpdates();
+        waitNoPendingTasksOnAll(); // FIXME why do I need this to see the new mappings?
+
+        assertMappings(TestSystemIndexDescriptor.getNewMappings());
+    }
+
+    /**
+     * Check that if the the SystemIndexManager finds a managed index with mappings that claim to be newer than
+     * what it expects, then those mappings are left alone.
+     */
+    public void testSystemIndexManagerLeavesNewerMappingsAlone() throws Exception {
+        TestSystemIndexDescriptor.useNewMappings.set(true);
+
+        internalCluster().startNodes(1);
+        // Trigger the creation of the system index
+        assertAcked(prepareCreate(INDEX_NAME));
+        ensureGreen(INDEX_NAME);
+
+        assertMappings(TestSystemIndexDescriptor.getNewMappings());
+
+        // Poke the test descriptor so that the mappings are now out-dated.
+        TestSystemIndexDescriptor.useNewMappings.set(false);
+
+        // Cause a cluster state update, so that the SystemIndexManager will update the mappings in our index
+        triggerClusterStateUpdates();
+        waitNoPendingTasksOnAll(); // FIXME why do I need this to see the new mappings?
+
+        // Mappings should be unchanged.
+        assertMappings(TestSystemIndexDescriptor.getNewMappings());
+    }
+
+    /**
+     * Performs a no-op cluster state update in order to trigger any cluster state listeners - specifically, SystemIndexManager.
+     */
+    private void triggerClusterStateUpdates() {
```

Comment:
Can we do something else like put a template or an empty reroute command? I prefer not to reach into a node a grab services if there is a way to trigger via an API